packages/vite/src/node/optimizer/index.ts

```diff
@@ -242,7 +243,16 @@ export async function optimizeDeps(
     const flatId = flattenId(id)
     flatIdDeps[flatId] = deps[id]
     const entryContent = fs.readFileSync(deps[id], 'utf-8')
-    const exportsData = parse(entryContent) as ExportsData
+    let exportsData: ExportsData
+    try {
+      exportsData = parse(entryContent) as ExportsData
+    } catch {
+      config.logger.warn(
+        `Unable to parse dependency: ${id}. Trying again with an esbuild transform.`
```

Comment:
Probably shouldn't add a warning for something the user can't do anything to avoid

Reply:
Good point, used a debug statement instead 👍